.github/workflows/pytest.yml

```diff
@@ -0,0 +1,42 @@
+name: pytest
+
+on:
+  pull_request:
+    paths:
+      - "src/*"
+
+jobs:
+  pytest:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: "Fedora Rawhide"
+            image: "registry.fedoraproject.org/fedora:rawhide"
+          - name: "Fedora Latest"
+            image: "registry.fedoraproject.org/fedora:latest"
+          - name: "CentOS Stream 9"
+            image: "quay.io/centos/centos:stream9"
+          - name: "CentOS Stream 8"
+            image: "quay.io/centos/centos:stream8"
+
+    runs-on: "ubuntu-latest"
+    container:
+      image: ${{ matrix.image }}
+
+    steps:
+      - name: "Checkout the repository"
+        uses: actions/checkout@v4
+
+      - name: "Install dependencies"
+        # Setuptools is required because we still rely on 'distutils' which have been dropped in Python 3.12
```

Comment:
I guess this comment goes better in `requirements.txt` to document `setuptools` there; this way, once `distutils` is dropped, also its comment can go (and there is no change needed here)